pageserver/src/tenant/timeline.rs

```diff
@@ -1748,9 +1533,11 @@ impl Timeline {
         guard.initialize_local_layers(loaded_layers, disk_consistent_lsn + 1);
 
         if let Some(rtc) = self.remote_client.as_ref() {
+            let (needs_upload, needs_cleanup) = sync;
             for (layer, m) in needs_upload {
-                rtc.schedule_layer_file_upload(&layer.layer_desc().filename(), &m)?;
+                rtc.schedule_layer_file_upload(layer, &m)?;
             }
+            rtc.schedule_layer_file_deletion(&needs_cleanup)?;
```

Comment:
This is a rather big change; previously we leaked all of these, but I don't see why should we continue to because it was easier to spot with the new version. It will also be easier for John to switch all these to deletion queue.

Reply:
Big or small.

Moved to #5103.